actions.opy

```diff
@@ -0,0 +1,51 @@
+#!mainFile "main.opy"
+
+rule "Enemy is in sight - switch to attacking mode":
+    @Event eachPlayer
+    @Condition eventPlayer.isDummy()
+    @Condition eventPlayer.walkBehavior != BotWalkBehavior.FIGHTING
+    @Condition any([isInLoS(eventPlayer.getEyePosition(), p.getEyePosition(), BarrierLos.PASS_THROUGH_BARRIERS) for p in getPlayers(Team.1)])
+
+    # Get all the enemies that the bot sees.
+    eventPlayer.seenEnemies = [p for p in getPlayers(Team.1) if isInLoS(eventPlayer.getEyePosition(), p.getEyePosition(), BarrierLos.PASS_THROUGH_BARRIERS) and distance(eventPlayer.getPosition(), p) < 20]
+
+    # Select target based on threat/attractiveness, distance, role (dps are most likely to attack, then tank, then support), personal health, and maybe some other shit idk
+    # TODO: actually do those things lul
+    eventPlayer.targetPlayer = sorted(eventPlayer.seenEnemies, lambda p: p.attractiveness - p.threat).last()
+
+    smallMessage(getAllPlayers(), "Bot found target")
+
+    # TODO: evaluate whether it's actually worth it to go for the enemy
+
+    if entityExists(eventPlayer.targetPlayer):
```

Comment:
why would the player not exist? if it can, shouldn't we filter out wacky players in the sorted array

Reply:
gotta be honest i can't actually remember why i added this, i think it was because i was afraid there'd be a chance that seenEnemies is empty?